buildrunner/utils.py

```diff
@@ -116,6 +116,31 @@ def tempfile(prefix=None, suffix=None, temp_dir='/tmp'):
     return os.path.join(temp_dir, name)
 
 
+def checksum(files: list=None):
+    """
+    Generate a single SHA1 checksum of the list of files passed in
+    """
+    if not isinstance(files, list):
+        print(f"WARNING: TypeError - Input 'files' needs to be a list instead of {type(files)}")
+        sys.exit()
+
+    blocksize = 2 ** 16
+    hasher = hashlib.sha1()
+
+    for filename in sorted(files):
+        if not os.path.isfile(filename):
+            print(f"WARNING: {filename} does not exist, skipping file.")
+            continue
+
+        with open(filename, "rb") as open_file:
+            buf = open_file.read(blocksize)
+            while len(buf) > 0:
+                hasher.update(buf)
+                buf = open_file.read(blocksize)
```

Comment:
I'm curious why you decided to read this in blocks instead of all at once? Is this just to keep memory usage down?

Reply:
Honestly I just borrowed what was done below in `hash_sha1()`

Reply:
I should probably just use `hash_sha1()` in the jinja template if possible.

Reply:
I don't like that `def checksum(files: list=None):` is so similar to `def hash_sha1(file_name_globs=None):`. But I don't know how to pass in `file_name_globs` from the `template/buildrunner.yaml` and I didn't want to break any existing functionality that `def hash_sha1(file_name_globs=None):` has.

Reply:
Ditto. I knew why you did it and agreed with it, even though I didn't like it either :D